### PR TITLE
fix(NativeApps): Scroll to top after pushing route in MessageSync

### PR DESCRIPTION
**Pull request by patrickvenetz**
When app pushes a route, scroll to top afterwards.

### CODE REVIEW

**Review by tpreusse — APPROVED** [review]

**Comment by ovbm** [comment]
Did you test if this doesn't break navigating to a specific comment from a notification?

**Comment by patrickvenetz** [comment]
(Testing pending.)

**Comment by tpreusse** [comment]
Scroll to comment should be fine, we already wait 50ms before scrolling to the comment for other reasons:

https://github.com/orbiting/republik-frontend/blob/2e69fc4b1a36b832202f35d91d4212c6b17902bd/components/Discussion/Comments.js#L168-L175

But there is another issue. I'll add a suggestion. Thanks for the reminder of our scroll to cases @ovbm.

File: components/NativeApp/MessageSync.js [hunk 1]
```diff
@@ -108,7 +108,9 @@ const MessageSync = ({ upsertDevice, me, client }) => {
       } else if (content.type === 'authorization') {
         checkPendingAppSignIn()
       } else if (content.type === 'push-route') {
-        Router.pushRoute(content.url.replace(PUBLIC_BASE_URL, ''))
+        Router.pushRoute(content.url.replace(PUBLIC_BASE_URL, '')).then(() => {
+          window.scrollTo(0, 0)
+        })
```
Comment (tpreusse) [inline components/NativeApp/MessageSync.js:113]: There are potential `scrollTo` conflicts, to work around this we should back-off when there is a hash in the url:
```suggestion
        const targetUrl = content.url.replace(PUBLIC_BASE_URL, '')
        Router.pushRoute(targetUrl).then(() => {
          if (targetUrl.indexOf('#') === -1) {
            window.scrollTo(0, 0)
          }
        })
```

We should test opening `https://www.republik.ch/konto#newsletter`—used in newsletters—after deploy. Currently works on iOS most of the time.

I also just noticed that next 10 has already improved its scroll logic and seems to do scroll 0,0 by default now. But probably won't get around to do the update until late march (needs routing testing and potentially changes).

**Comment by ovbm** [comment]
Testing results:

Android: Fixes bug, now always on top of feed when navigating back.

iOS: Doesn't introduce regressions. Navigating back preserves previous feed position. Opening notifications with comment reply scrolls to respective comment within the dialogue.

**Pull request merged**